src/components/ebay-menu/index.js

```diff
@@ -250,7 +250,12 @@ function processAfterStateChange(itemIndexes) {
  * @param {HTMLElement} itemEl
  */
 function handleItemClick(e, itemEl) {
-    this.setCheckedItem(getItemElementIndex(itemEl), true);
+    const itemElIndex = getItemElementIndex(itemEl);
+    if (this.getCheckedList().indexOf(itemElIndex) === -1) {
```

Comment:
For this first conditional block, do we need to check that it's a radio and not a checkbox?

Also, are these code comments correct? It says getCheckedList is only for checkboxes, not radios.

```js
/**
 * Determine currently checked items (for checkbox case)
 * @returns {Array} checked indexes
 */
function getCheckedList() {
```

Reply:
>do we need to check that it's a radio and not a checkbox?

We have a logical case for emitting an event that is neither a `checkbox` or `radio` type, hence I didn't limit it to the radio use case: https://github.com/eBay/ebayui-core/blob/92c456f10fbfc0712d61a9e551db83e68e74af09/src/components/ebay-menu/index.js#L242-L243

>are these code comments correct?

Probably not! Let me review the comment throughout.

Reply:
Ok, it's a little tricky to follow. I agree with @DylanPiercey that this component would benefit from a refactor when we go to MarkoV4. I think when we separate it from button (#713) is going to help too.

Reply:
Yes, for sure. It didn't get refactored so it has lots of "ins and outs".